### PR TITLE
[WIP] Add a like button

### PR DESCRIPTION
**Pull request by StCyr**
* Resolves: https://github.com/nextcloud/social/issues/617
* Target version: master 

### Summary

The like button is created but I've got various server-side errors here:

1. When I like a random post, I'm getting: `{"status":-1,"exception":"OCA\\Social\\Exceptions\\StreamNotFoundException","message":"Stream is not Public"}`. Note that I get the same error when I try to boost a random post;
2. When I like a post of mine, I'm getting: `{"status":-1,"exception":"Doctrine\\DBAL\\Exception\\TableNotFoundException","message":"An exception occurred while executing 'SELECT `a`.`id`, `a`.`type`, `a`.`actor_id`, `a`.`object_id`, `a`.`creation` FROM `oc_social_a2_actions` `a` WHERE (LOWER(`a`.`actor_id`) = LOWER(?)) AND (LOWER(`a`.`object_id`) = LOWER(?)) AND (`a`.`type` = ?)' with params [\"https:\\\/\\\/nextcloud.acme.org\\\/index.php\\\/apps\\\/social\\\/@cyrille\", \"https:\\\/\\\/nextcloud.acme.org\\\/index.php\\\/apps\\\/social\\\/@cyrille\\\/15625980311989442625\", \"Like\"]:\n\nSQLSTATE[42S02]: Base table or view not found: 1146 Table 'nextcloud.oc_social_a2_actions' doesn't exist"}`

### CODE REVIEW

**Comment by violoncelloCH** [comment]
ahh, the sign-off is missing here as well @StCyr, could you add it?

regarding the errors: try setting `occ config:app:set --value '0.1.0' social installed_version`. This triggers the DB migrations and fixed the errors for me :)

**Comment by StCyr** [comment]
> ahh, the sign-off is missing here as well @StCyr, could you add it?
> 

This is harder here as there are already 3 commits. Should I make a new PR?

**Comment by violoncelloCH** [comment]
in the "checks" section right above the merge button there is DCO; when you click on details there, it gives some nice instructions :)
but if it's easier for you, you can also create a new PR...

**Comment by daita** [comment]
I merged the fix on the three-dot-button, now we have conflicts. Can you rebase and fix them ?

>     1. When I like a random post, I'm getting: `{"status":-1,"exception":"OCA\\Social\\Exceptions\\StreamNotFoundException","message":"Stream is not Public"}`. Note that I get the same error when I try to boost a random post;

Let's assume you can like any Post, I will remove the check in the back-end

**Comment by StCyr** [comment]
> I merged the fix on the three-dot-button, now we have conflicts. Can you rebase and fix them ?
> 

I've merged the changes. Is it ok?

File: src/store/timeline.js [hunk 1]
```diff
@@ -113,6 +129,30 @@ const actions = {
...
+				console.error('Failed to like post', error.response)
+				reject(error)
+			})
+		})
+	},
+	postUnlike(context, { post, parentAnnounce }) {
+		return axios.delete(OC.generateUrl(`apps/social/api/v1/post/like?postId=${post.id}`)).then((response) => {
+			context.commit('unlikePost', { post, parentAnnounce })
+			// eslint-disable-next-line no-console
+			console.log('Post unliked with token ' + response.data.result.token)
```
Comment (juliushaertl) [inline src/store/timeline.js]: console.debug should be usable without ignoring it :wink:

**Comment by daita** [comment]
good for me; maybe some check from @nextcloud/designers ?

Selection_087

(remember it's alpha, improvement can still be done later !)

**Comment by daita** [comment]
merged, comments about the UI can be done on a BR

**Pull request merged**